modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+|`initialResource` |YAML/Object representing a resource to be shown by
+the editor. This prop is used only during the inital render
+
+|`header` |Add a header on top of the YAML editor
+
+|`onSave` |Callback for the Save button. Passing it will override the
+default update performed on the resource by the editor
+|===
+
+[discrete]
+== `ResourceEventStream`
+
+A component to show events related to a particular resource.
+
+[source,tsx]
+----
+const [resource, loaded, loadError] = useK8sWatchResource(clusterResource);
+return <ResourceEventStream resource={resource} />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resource` |An object whose related events should be shown.
+|===
+
+[discrete]
+== `usePrometheusPoll`
+
+Sets up a poll to Prometheus for a single query. It returns a tuple containing the query response, a boolean flag indicating whether the response has completed, and any errors encountered during the request or post-processing of the request.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`` |\{PrometheusEndpoint} props.endpoint - one of the
+PrometheusEndpoint (label, query, range, rules, targets)
+
+|`` |\{string} [props.query] - (optional) Prometheus query string. If
+empty or undefined, polling is not started.
+
+|`` |\{number} [props.delay] - (optional) polling delay interval (ms)
+
+|`` |\{number} [props.endTime] - (optional) for QUERY_RANGE enpoint, end
+of the query range
+
+|`` |\{number} [props.samples] - (optional) for QUERY_RANGE enpoint
+
+|`` |\{number} [options.timespan] - (optional) for QUERY_RANGE enpoint
+
+|`` |\{string} [options.namespace] - (optional) a search param to append
+
+|`` |\{string} [options.timeout] - (optional) a search param to append
+|===
+
+[discrete]
+== `Timestamp`
+
+A component to render timestamp. The timestamps are synchronized between invidual instances of the Timestamp component. The provided timestamp is formatted according to user locale.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`timestamp` |the timestamp to render. Format is expected to be ISO 8601
+(used by Kubernetes), epoch timestamp, or an instance of a Date.
+
+|`simple` |render simple version of the component omitting icon and
+tooltip.
+
+|`omitSuffix` |formats the date ommiting the suffix.
+
+|`className` |additional class name for the component.
+|===
+
+[discrete]
+== `useModal`
+
+A hook to launch Modals.`tsx<br/>const AppPage: React.FC = () => {<br/> const [launchModal] = useModal();<br/> const onClick = () => launchModal(ModalComponent);<br/> return (<br/>   <Button onClick={onClick}>Launch a Modal</Button><br/> )<br/>}<br/>`
+
+[discrete]
+== `ActionServiceProvider`
+
+Component that allows to receive contributions from other plugins for the `console.action/provider` extension type. See docs:
+https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/docs/console-extensions.md#consoleactionprovider
```

Comment:
Is this doc approved to include in the end user docs?

Reply:
hmm probably not

Reply:
left text but removed link